File: packages/ts/src/strategies/fix-keys.ts
/**
 * fix-keys: Quote bare (unquoted) object keys.
 * Must-isolate: runs after fix-quotes.
 * Handles simple identifiers only — does not attempt to re-parse.
 */
export const fixKeys = (text: string): string => {
  // Match: start of object value — a bare identifier followed by colon
  // Negative lookbehind: not already in a string context (simplified heuristic)
  // Pattern: word chars at start of key position (after { , or newline)
  return text.replace(
    /([{,\n\r]\s*)([A-Za-z_$][\w$]*)(\s*):/g,
    (_, prefix: string, key: string, space: string) =>
      `${prefix}"${key}"${space}:`,
  );
};

File: packages/python/src/output_guard/strategies/fix_keys.py
"""fix-keys: Quote bare (unquoted) object keys.

Mirror of packages/ts/src/strategies/fix-keys.ts.
"""

from __future__ import annotations

import re

_BARE_KEY = re.compile(r"([{,\n\r]\s*)([A-Za-z_$][\w$]*)(\s*):")


def fix_keys(text: str) -> str:
    return _BARE_KEY.sub(r'\1"\2"\3:', text)
